### PR TITLE
clickhouse: add supported table functions and file formats

### PR DESCRIPTION
**Pull request by wojcik-dorota**
https://aiven.atlassian.net/browse/DOC-552

### CODE REVIEW

File: docs/products/clickhouse/reference/table-functions.rst
Comment (ArthurAnsquer) [inline docs/products/clickhouse/reference/table-functions.rst:1]: Maybe this page could be called "supported-table-functions" and has a similar title, as we currently have a "supported-table-engines" page (also we have the above supported-file-formats)! To keep it consistent, what do you think?

File: docs/products/clickhouse/reference/table-functions.rst [hunk 1]
```diff
@@ -0,0 +1,31 @@
+Table functions supported in Aiven for ClickHouse®
+==================================================
+
+Table functions return data of the table type and are used to construct tables. Aiven for ClickHouse® supports the following `table functions <https://clickhouse.com/docs/en/sql-reference/table-functions>`_:
+
+* ``cluster``
+* ``clusterAllReplicas``
+* ``cosn``
+* ``deltaLake``
```
Comment (ArthurAnsquer) [inline docs/products/clickhouse/reference/table-functions.rst:9]: @joelynch I omitted dictionary on purpose given what you told me on slack! We can modify the doc later down the line when we work more on the feature :)

**Review by ArthurAnsquer — COMMENTED** [review]
Looking fine! Suggested title change otherwise all good! @joelynch could you double check and merge if it looks fine to you? I don't have the permissions to do so yet :D

File: docs/products/clickhouse/reference/s3-supported-file-formats.rst [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+The `S3 table function <https://clickhouse.com/docs/en/sql-reference/table-functions/s3>`_ allows you to select and insert data in S3-compatible storages. The S3 table function in Aiven for ClickHouse® can be used with the following file formats:
+
+* ``Arrow``
+* ``CSV``
+* ``JSON``
+* ``TSV``
+* ``Parquet``
+* ``ORC``
+* ``Avro``
```
Comment (joelynch) [inline docs/products/clickhouse/reference/s3-supported-file-formats.rst:12]: These are just the file formats I tested, we support a lot more than just this but should be good for the moment

**Review by joelynch — APPROVED** [review]
Looks good!  Added a suggestion

File: docs/products/clickhouse/reference/table-functions.rst
Comment (wojcik-dorota) [reply docs/products/clickhouse/reference/table-functions.rst:1]: updated the file name but I think we should keep the titles as they are - adding a nice-to-have "supported" word is ok if the title is short. For the loner titles, I wouldn't make them even longer by adding "supported", even for consistency.

**Review by joelynch — APPROVED** [review]

**Comment by joelynch** [comment]
@wojcik-dorota do you want me to merge this?  Is the author allowed to merge for this repo?

**Comment by wojcik-dorota** [comment]
> @wojcik-dorota do you want me to merge this? Is the author allowed to merge for this repo?

@joelynch the author can merge so I'll do that. Thanks for all your support :-)

**Pull request merged**